Hash-Tables-and-Hashing Functions.md

```diff
@@ -0,0 +1,150 @@
+# Hash Tables and Hashing Functions
+
+## Introduction to hashing
+
+Hashing is designed to solve the problem of needing to efficiently find or store an item in a collection. 
+For example, if we have a list of 10,000 words of English and we want to check if a given word is in the list, it would be inefficient to successively compare the word with all 10,000 items until we find a match. 
```

Comment:
Maybe mention here in passing that even though they are sorted in alphabetical order, like in a dictionary, you still need some time to search for it.